bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/scheduler/AbstractExpression.java

```diff
@@ -240,11 +272,36 @@ public Date getFinalFireTime() {
             }
         }
 
-        if (getCandidates().isEmpty()) {
-            return null;
+        Date lastCandidate = null;
+
+        if (!getCandidates().isEmpty()) {
+            if (getCandidates().size() == 1) {
+                lastCandidate = getCandidates().get(0);
+            } else {
+
+                while (getCandidates().size() == maximumCandidates) {
```

Comment:
If we used a floating start date, couldn't the size be equal to maximum candidates, too?
You set the minimum number in the constructor to 0 for CronExpression, but is this a grant, that there aren't the maximum number be used?